### ISSUE TITLE
StringComparer.InvariantCultureIgnoreCase and string.ToUpperInvariant returns inconsistent result for subscript r.

### ISSUE THREAD

**Issue by faddiv** [dotnet/runtime#1298]
# Details

I was experimenting with my natural string comparer implementation when I noticed the following oddity with StringComparer.InvariantCultureIgnoreCase. 
If I compare "ᵣ" and "ʀ" then I got "ᵣ" < "ʀ"
If I compare "ᵣ" and "Ʀ" then I got "ᵣ" = "Ʀ"
If I compare "Ʀ" and "ʀ" then I got "Ʀ" < "ʀ" 
Meanwhile:
If I call "ʀ".ToUpperInvariant() then I got "Ʀ".
and if I call "ᵣ".ToUpperInvariant() then it didn't change and remains "ᵣ".
So these can't used consistently even if they are use the same invariant culture.

# Environment
Platform: Windows 10 Pro version 1903 OS build: 18362.535
Tested under .net core 2.2 and .net core 3.1

# XUnit tests
These are the unit tests I experimented with:

        [Fact]
        public void Uvular_trill_does_not_equal_to_subscript_r_if_case_ignored()
        {
            var invariantIgnoreCase = StringComparer.InvariantCultureIgnoreCase;
         
            invariantIgnoreCase.Compare("ᵣ", "ʀ").Should().NotBe(0);

            invariantIgnoreCase.Compare("ᵣ", "Ʀ").Should().NotBe(0);

            invariantIgnoreCase.Compare("Ʀ", "ʀ").Should().Be(0);
        }

        [Fact]
        public void ToUpperInvariant_works_consistently_with_InvariantCultureIgnoreCase()
        {
            var invariantIgnoreCase = StringComparer.InvariantCultureIgnoreCase;

            var allCharacter = new string[char.MaxValue - char.MinValue];
            for (char ch1 = char.MinValue; ch1 < char.MaxValue; ch1++)
            {
                allCharacter[ch1 - char.MinValue] = $"{ch1}";
            }
            var actual = allCharacter
                .Select(character => new { character, code = (int)character[0] })
                .OrderBy(e => e.character, invariantIgnoreCase)
                .ThenBy(e => e.code)
                .ToList();
            var expected = allCharacter
                .Select(character => new { character = character.ToUpperInvariant(), code = (int)character[0] })
                .OrderBy(e => e.character, invariantIgnoreCase)
                .ThenBy(e => e.code)
                .ToList();
            int i = 0;
            for (; i < allCharacter.Length; i++)
            {
                if (actual[i].code != expected[i].code)
                    break;
            }
            if (i == allCharacter.Length)
                return;
            actual[i].Should().Be(expected[i],
                $"At index {i} the same character should be");
        }

**Comment by scalablecory** [comment]
@GrabYourPitchforks @tarekgh

**Comment by Gnbrkm41** [comment]
Those looking like the same doesn't really matter; what matters is the relationship between those characters defined in the Unicode standard.
```
ᵣ U+1D63 Latin Subscript Small Letter R
ʀ U+0280 Latin Letter Small Capital R
Ʀ U+01A6 Latin Letter Yr
```
U+1D63 having no 'upper-case' equivalent, compares different from the other two with IgnoreCase and remains the same after the ToUpperInvariant call; whereas U+0280 and U+01A6 are mapped as 'lowercase' and the 'uppercase' pair and therefore does compare equal with each other with IgnoreCase and returns different character.

**Comment by tarekgh** [comment]
@faddiv looking at Unicode, 

```
01A6;LATIN LETTER YR;Lu;0;L;;;;;N;LATIN LETTER Y R;;;0280;
0280;LATIN LETTER SMALL CAPITAL R;Ll;0;L;;;;;N;;;01A6;;01A6
```

This means the casing of `ᵣ` mapped to `ʀ` is correct. 

The only problem now is why "ᵣ" = "Ʀ" (codepoint U+1D63)?  I checked that and I found this is true on Windows but not on Linux. Net Core gets the behavior form the underlying OS so this issue looks a Windows-specific issue and needs to be fixed in Windows if agreed on. I am going to close this issue as we cannot fix it in the .Net core and the fix has to come from Windows. I'll send this problem to the Windows team but feel free to log an issue for them too through Windows 10 Feedback Hub. Thanks.
